### PR TITLE
update RBAC with more detailed permission

### LINKED ISSUE #687: Ensure that the federated control plane is minimally privileged
#670 revealed that the helm chart was not granting sufficient privileges to the control plane.  While it is tempting to just use the cluster-admin role, it should be possible to grant only the privileges required.  For some types, the scope of privilege will depend on how the control plane is deployed (namespaced or cluster-scoped):

 - `get/watch/list/update`
   - `(multiclusterdns|scheduling|types).federation.k8s.io` 
      - namespaced: federation namespace
      - cluster-scoped: *
   - `FederatedTypeConfig` 
       - federation namespace
   - `FederatedCluster` 
       - federation namespace
   - `PropagatedVersion` 
       - namespaced: federation namespace
       - cluster-scoped: *
   - `ClusterPropagatedVersion` 
       - cluster-scoped: *
 - `get/list/watch`
   - `Namespace`
      - cluster-scoped: *
 - `get`
   - `Cluster` 
       - cluster registry namespace
   - `Secret` (for cluster auth)
       - federation namespace

For member clusters, the permissions required by the local service account would depend on the enabled types:

 - `get/watch/list/update`
   - any type intended to be managed by federation
      - namespaced: federation namespace
      - cluster-scoped: *

If the goal was avoiding the use of the cluster-admin role in member clusters, it would be necessary to coordinate the rbac permissions of member cluster service accounts with the types enabled for federation.

cc: @xunpan @pmorie

### PR DESCRIPTION
**Pull request by xunpan**
Fixes #687

### CODE REVIEW

File: charts/federation-v2/charts/controllermanager/templates/clusterrole.yaml [hunk 1]
```diff
@@ -56,15 +56,34 @@ rules:
 - apiGroups:
   - ""
   resources:
-  - configmaps
-  - secrets
-  - events
   - namespaces
```
[the remark this answers is not in the document]
Comment (xunpan) [reply charts/federation-v2/charts/controllermanager/templates/clusterrole.yaml]: addressed

**Comment by marun** [comment]
I think this is almost there, but as per my last review comment I don't think the control plane needs cluster-wide permissions for configmaps and secrets if it only needs to access those resources in the federation namespace.  Would it make sense to add a new role (not cluster role) and role binding granting those localized permissions in this PR?

**Comment by marun** [comment]
@xunpan I'll merge this and we can fix the permissions in a follow-up.

/approve
/lgtm

**Comment by xunpan** [comment]
> I think this is almost there, but as per my last review comment I don't think the control plane needs cluster-wide permissions for configmaps and secrets if it only needs to access those resources in the federation namespace. Would it make sense to add a new role (not cluster role) and role binding granting those localized permissions in this PR?

I will handle this.

**Pull request merged**